axum-extra/src/extract/host.rs

```diff
@@ -59,6 +60,24 @@ where
     }
 }
 
+impl<S> OptionalFromRequestParts<S> for Host
+where
+    S: Send + Sync,
+{
+    type Rejection = Infallible;
+
+    async fn from_request_parts(
+        parts: &mut Parts,
+        _state: &S,
+    ) -> Result<Option<Self>, Self::Rejection> {
+        Ok(
+            <Self as FromRequestParts<S>>::from_request_parts(parts, _state)
+                .await
+                .ok(),
+        )
```

Comment:
if I understand correctly, this implementation ignores any rejections and turns them into `None`, which would match the previous behavior. I'm not sure if that's the behavior we want though.

it might be better to turn `OptionalFromRequestParts` into the primary implementation and in `FromRequestParts` we use `parts.extract::<Option<Host>>().await.unwrap_or(FailedToResolveHost)` or something roughly like that.

Reply:
Had to `.ok().flatten()` it because it's a `Result<Option<Host>, Infallible>` now